lstchain/io/dataselection.py

```diff
@@ -0,0 +1,227 @@
+        default_value=2.83,
+    ).tag(config=True)
+
+    lst_tel_ids = List(
+        help="List of selected LST telescope ids",
+        trait=Int(),
+        default_value=[1],
+    ).tag(config=True)
+
+    def filter_cut(self, data):
+        return filter_events(data, self.event_filters())
+
+    def gh_cut(self, data):
+        return data[data["gh_score"] > self.fixed_gh_cut]
+
+    def theta_cut(self, data):
+        return data[data["theta"] < u.Quantity(self.fixed_theta_cut) * u.deg]
+
+    def true_src_fov_offset_cut(self, data):
+        return data[
+                data["true_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut) * u.deg
+            ]
+
+    def reco_src_fov_offset_cut(self, data):
+        return data[
+                data["reco_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut) * u.deg
+            ]
+
+    def tel_ids_filter(self, data):
+        for i in self.lst_tel_ids:
+            data["sel_tel"] = data["tel_id"] == i
+        return data[data["sel_tel"]]
+
+    def event_filters(self):
+        """
+        Creates a dict for filter_events() function
+        """
+        self.evt_filter = {
+            "intensity": self.intensity,
+            "width": self.width,
+            "length": self.length,
+            "r": self.r,
+            "wl": self.wl,
+            "leakage_intensity_width_2": self.leakage_intensity_width_2,
+        }
+        return self.evt_filter
+
+
+class DataBinning(Component):
+    """
+    Collects information on generating energy and angular bins for
+    generating IRFs as per pyIRF requirements.
+    """
+
+    true_energy_bins = List(
+        help="Values to get true energy (TeV) binning as "
+        "[e_min, e_max, bins_per_decade]",
+        trait=Float(),
+        default_value=[0.01, 100, 5.5],
+    ).tag(config=True)
+
+    reco_energy_bins = List(
+        help="Values to get reco energy (TeV) binning as "
+        "[e_min, e_max, bins_per_decade]",
+        trait=Float(),
+        default_value=[0.01, 100, 5.5],
+    ).tag(config=True)
+
+    energy_migra_bins = List(
+        help="Values to get energy migration binning as [min, max, bins]",
+        default_value=[0.2, 5, 31],
+    ).tag(config=True)
+
+    fov_offset_bins = List(
+        help="List of bins for FOV offset binning",
+        trait=Float(),
+        default_value=[0.3, 0.5, 0.7],
+    ).tag(config=True)
+
+    bkg_fov_offset_bins = List(
+        help="Range of values for multiple FOV offset binning "
+        "for Background IRF as [o_min, o_max]",
+        trait=Float(),
+        default_value=[0, 11],
+    ).tag(config=True)
+
+    source_offset_bins = List(
+        help="Values to get source offset binning for PSF IRF "
+        "as [o_min, o_max, bin_width]",
+        trait=Float(),
+        default_value=[0, 1.0001, 0.001],
+    ).tag(config=True)
+
+    def true_energy(self):
```

Comment:
You overwrite the method with the bins array in the method here. If you want to avoid calculating the bins twice, you can use astropy's `@lazyproperty` returning the value.

But a normal `@property` should also be fine.

Just don't overwrite the method with the bin array.

Reply:
Ok